src/Common/src/System/Data/ProviderBase/DbMetaDataFactory.cs

```diff
@@ -171,7 +171,7 @@ private DataTable ExecuteCommand(DataRow requestedCollectionRow, string[] restri
                 schemaTable = reader.GetSchemaTable();
                 foreach (DataRow row in schemaTable.Rows)
                 {
-                    resultTable.Columns.Add(row["ColumnName"] as string, (Type)row["DataType"] as Type);
+                    resultTable.Columns.Add(row["ColumnName"] as string, row["DataType"] as Type);
```

Comment:
According to sharplab.io the hard cast has higher binding, so if this was wrong it was producing an InvalidCastException... it's probably better to leave the hard cast and remove the as-cast (since that's the "functionally the same but less confusing" change)